torchreid/data/datasets/image/classification.py

```diff
@@ -28,25 +27,14 @@ def __init__(self, root='', mode='train', **kwargs):
         ]
         self.check_before_run(required_files)
 
-        if mode == 'train':
-            train, classes = self.load_annotation(
-                self.annot,
-                self.data_dir,
-            )
-            test = []
-
-        elif mode == 'test':
-            test, classes = self.load_annotation(
-                self.annot,
-                self.data_dir,
-            )
-            train = []
+        data, classes = self.load_annotation(
+            self.annot,
+            self.data_dir,
+        )
 
-        else:
-            classes = []
-            train, test = [], []
-
-        super().__init__(train, test, mode=mode, classes=classes, **kwargs)
+        super().__init__(data, **kwargs)
+        self.classes = classes
```

Comment:
Why don't you pass classes and num_ids as args to constructor?

Reply:
good point!

Reply:
done